tensorstore/kvstore/s3/s3_key_value_store.cc

```diff
@@ -79,14 +80,18 @@
 #include "tensorstore/util/result.h"
 #include "tensorstore/util/status.h"
 #include "tensorstore/util/str_cat.h"
-
 // specializations
 #include "tensorstore/internal/cache_key/std_optional.h"  // IWYU pragma: keep
 #include "tensorstore/internal/json_binding/std_array.h"  // IWYU pragma: keep
 #include "tensorstore/internal/json_binding/std_optional.h"  // IWYU pragma: keep
 #include "tensorstore/serialization/fwd.h"  // IWYU pragma: keep
 #include "tensorstore/serialization/std_optional.h"  // IWYU pragma: keep
 #include "tensorstore/util/garbage_collection/std_optional.h"  // IWYU pragma: keep
+#include "tensorstore/internal/json_binding/bindable.h"
+#include "tensorstore/internal/poly/poly_impl.h"
```

Comment:
iwyu doesn't handle poly well; remove poly/poly_impl / poly/storage  here.

Reply:
In fact, I think that none of these edits are needed.

When running IWYU, double check where include edits happen & keep them in the main-block above, this section is for specific specializations that are ADL-used, not directly used.

Reply:
removed